In [2]:
import pandas as pd  # pandas
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

# Adatbázis séma (Diagram)

<img src="table_diagram.png">

# Adattáblák

## events

In [6]:
events = pd.read_csv("events.csv", sep=",")

In [9]:
events.head()

event_id            device_id            timestamp  longitude  latitude
0         1    29182687948017175  2016-05-01 00:55:25     121.38     31.24
1         2 -6401643145415154744  2016-05-01 00:54:12     103.65     30.97
2         3 -4833982096941402721  2016-05-01 00:08:05     106.60     29.70
3         4 -6815121365017318426  2016-05-01 00:06:40     104.27     23.28
4         5 -5373797595892518570  2016-05-01 00:07:18     115.88     28.66

In [10]:
events.count()

event_id     3252950
device_id    3252950
timestamp    3252950
longitude    3252950
latitude     3252950
dtype: int64

## gender_age_train

In [12]:
gender_age_train = pd.read_csv("gender_age_train.csv", sep=","")

In [13]:
gender_age_train.head()

device_id gender  age   group
0 -8076087639492063270      M   35  M32-38
1 -2897161552818060146      M   35  M32-38
2 -8260683887967679142      M   35  M32-38
3 -4938849341048082022      M   30  M29-31
4   245133531816851882      M   30  M29-31

In [14]:
gender_age_train.count()

device_id    74645
gender       74645
age          74645
group        74645
dtype: int64

In [17]:
gender_age_train.groupby('group')['device_id'].count()

group
F23-      5050
F24-26    4190
F27-28    3118
F29-32    4628
F33-42    5561
F43+      4194
M22-      7488
M23-26    9605
M27-28    5445
M29-31    7309
M32-38    9476
M39+      8581
Name: device_id, dtype: int64

In [19]:
gender_age_train.groupby('gender')['device_id'].count()

gender
F    26741
M    47904
Name: device_id, dtype: int64

## phone_brand_device_model

In [45]:
phone_brand_device_model = pd.read_csv("phone_brand_device_model.csv", sep=",")

In [46]:
phone_brand_device_model.head()

device_id phone_brand   device_model
0 -8890648629457979026          小米             红米
1  1277779817574759137          小米           MI 2
2  5137427614288105724          三星      Galaxy S4
3  3669464369358936369       SUGAR           时尚手机
4 -5019277647504317457          三星  Galaxy Note 2

In [ ]:
# a kínai márkákat lefordítjuk...

In [50]:
translate = pd.read_csv("translate.csv", sep=",")

In [51]:
translate.head()

phone_brand phone_brand_eng
0          三星         samsung
1          天语          Ktouch
2          海信         hisense
3          联想          lenovo
4          欧比             obi

In [55]:
phone_brand_device_model= phone_brand_device_model.merge(translate,on='phone_brand',how='right')

In [59]:
phone_brand_device_model.head()

device_id phone_brand device_model phone_brand_eng
0 -8890648629457979026          小米           红米          Xiaomi
1  1277779817574759137          小米         MI 2          Xiaomi
2 -3883532755183027260          小米        MI 2S          Xiaomi
3 -5827952925479472594          小米  MI One Plus          Xiaomi
4 -5776341595519925628          小米        MI 2A          Xiaomi

## app_events

In [60]:
app_events = pd.read_csv("app_events.csv", sep=",")

In [61]:
app_events.head()

event_id               app_id  is_installed  is_active
0         2  5927333115845830913             1          1
1         2 -5720078949152207372             1          0
2         2 -1633887856876571208             1          0
3         2  -653184325010919369             1          1
4         2  8693964245073640147             1          1

In [62]:
app_events.count()

event_id        32473067
app_id          32473067
is_installed    32473067
is_active       32473067
dtype: int64

In [69]:
app_events.groupby('event_id')['app_id'].agg('count').count()

1488096

In [70]:
# osszesen 1488096 db esemény van, ahhoz van 32473067 alkalmazás esemény 

## app_labels

In [71]:
app_labels = pd.read_csv("app_labels.csv", sep=",")

In [72]:
app_labels.head()

app_id  label_id
0  7324884708820027918       251
1 -4494216993218550286       251
2  6058196446775239644       406
3  6058196446775239644       407
4  8694625920731541625       406

In [73]:
app_labels.count()

app_id      459943
label_id    459943
dtype: int64

## label_categories

In [76]:
label_categories = pd.read_csv("label_categories.csv", sep=",")

In [77]:
label_categories.head()

label_id           category
0         1                NaN
1         2     game-game type
2         3   game-Game themes
3         4     game-Art Style
4         5  game-Leisure time

In [78]:
label_categories.count()

label_id    930
category    927
dtype: int64

In [ ]:
# itt van 3 sor NaN. Igazából szerintem nem számít.

# Adatok előkészítése

Szerintem feladatok:
    - a gender_age_train össze kellene join-olni a phone_brand_device_modellel
    - ugyancsak a gender_age_train-t össze kell join-olni az events-app_events-app_labels -en keresztül a label_categories-vel
    - a label kategória dummy változó lenne. A problémám, hogy itt akkro  930 db lenne...
    - bemenő változók: márka, alkalmazás label dummy. Célváltozó: group
